### PR TITLE
Implement LineString simplification

### PR DESCRIPTION
**Pull request by urschrei**
This trait currently uses the [Ramer–Douglas–Peucker](https://en.wikipedia.org/wiki/Ramer–Douglas–Peucker_algorithm) algorithm to
simplify a LineString (see #54), while preserving its overall shape.

This is WIP, for two reasons:
- `rdp()` currently uses `unwrap()`, because the `first()` and `last()` methods on slices return `Option`. It's not immediately clear to me how these could fail in this context, but all input is welcome.
- `point_line_distance()` and `rdp()` are implemented as helper functions, and called by the `Simplify` trait. This is mainly because `rdp()` is recursive and uses sub-slices of its initial input, which AFAICS can't be easily obtained from a `LineString`.

### CODE REVIEW

File: src/algorithm/simplify.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    #[test]
+    fn perpdistance_test() {
+        let start = Point::new(1.0, 2.0);
+        let end = Point::new(3.0, 4.0);
+        let p = Point::new(1.0, 1.0);
+        let dist = point_line_distance(&p, &start, &end);
+        assert_eq!(dist, 0.7071067811865475);
+    }
+    #[test]
+    fn rdp_test() {
```
Comment (frewsxcv) [inline src/algorithm/simplify.rs]: Can you also add a test doing `simplify` on an empty `LineString`?

Comment (frewsxcv) [inline src/algorithm/simplify.rs]: Or should we expect all `LineString`s to have at least two points? If so, we should make it explicit in the doc comment and add an `assert`.

File: src/algorithm/simplify.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+fn rdp<T>(points: &[Point<T>], epsilon: &T) -> Vec<Point<T>> 
+    where T: Float
+{
+    let mut dmax = T::zero();
+    let mut index: usize = 0;
+    let mut distance: T;
+
+    for (i, _) in points.iter().enumerate().take(points.len() - 1).skip(1) {
+        distance = point_line_distance(&points[i],
+                                       &*points.first().unwrap(),
```
Comment (frewsxcv) [inline src/algorithm/simplify.rs]: You might as well do `points[0]` here, right?

File: src/algorithm/simplify.rs (hunk 1 above)
Comment (urschrei) [inline src/algorithm/simplify.rs]: Hmm. A 2-point LineString can't be simplified anyway, and both it and an empty `LineString` are valid, so it makes sense to me to just return them both unmodified.

File: src/algorithm/simplify.rs (hunk 2 above)
Comment (urschrei) [inline src/algorithm/simplify.rs]: Instead of `.first()`? Yep, I just instinctively shy away from array indexing, but we know it'll be OK in this case.

**Comment by frewsxcv** [comment]
Everything here looks good to me, is this still a WIP?

**Comment by urschrei** [comment]
No, I'm pretty happy with it. How do you feel about the thoroughness of the tests, though? I feel like some tests for pathological input (tiny epsilon, gigantic LineString) would probably be a good idea at some point, although in the latter case that should probably happen further up the chain.

**Comment by frewsxcv** [comment]
Sounds great, thanks for this! 🎊

**Comment by urschrei** [comment]
\o/

**Comment by frewsxcv** [comment]
:D

One of these I want to get some sort of basic Rust GUI for geo data (maybe written in [conrod](https://github.com/PistonDevelopers/conrod)) so that we can visualize all these rust-geo algorithms and operations.
